src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -951,5 +961,17 @@ public virtual IEnumerable<string> GetDirectories(string path)
                        select p.Name;
             });
         }
+
+        private static bool IsReferenceUnavailableException(Exception e)
+        {
+            var comException = e as COMException;
+
+            if (comException == null)
+            {
+                return false;
+            }
+
+            return comException.HResult == unchecked((int)0x80004005);
```

Comment:
If [`VSLangProj.References.Add(...)`](https://msdn.microsoft.com/en-us/library/vslangproj.references.add.aspx) fails, it throws a COMException with the above HRESULT and the message "Reference unavailable."  MSDN documentation doesn't document this level of detail.